integration-commons/src/main/java/org/symphonyoss/integration/logging/IntegrationBridgeKeyProvider.java

```
@@ -0,0 +1,207 @@
+  }
+
+  /**
+   * Returns the session key to log into the cloud.
+   * It's going to verify if the session was created, if not will try to authenticate.
+   * @return Session token
+   */
+  @Override
+  public String getSessionKey() {
+    String sReturn;
+
+    try {
+      sReturn = authProvider.getSymphonyAuth().getSession();
+
+      if (sReturn == null) {
+        //the following is for lazy authentication in a separate thread while we are waiting for
+        // the pod to come up.
+        //Until we can return a valid session key value, the cloud logger will keep log messages
+        //in an internal queue. As soon as authentication is complete and session key can be
+        //returned, the messages will be pushed to the cloud.
+
+        //This method may be called by multiple threads. Don't start authentication multiple times -
+        //only if we haven't run this or if it has already finished then fire a new one
+        synchronized (IntegrationBridgeCloudLoggerFactory.class) {
+          if ((authenticationFuture == null) ||
+              ((authenticationFuture != null) && (authenticationFuture.isDone()))) {
+
+            authenticationFuture = executorForAuthentication.submit(new Runnable() {
+              @Override
+              public void run() {
+                try {
+                  //wait a minute before trying - otherwise authentication code
+                  //logs too many exceptions
+                  Thread.sleep(60000);
+                  reAuth();
+                } catch (Exception e) {
+                  //there will be multiple exceptions during pod start up. They do not indicate real
+                  //problems - simply not all components are up and running yet. Hence the log level
+                  //"info" and only exception message, not the full stacktrace
+                }
+              }
+            });
+          }
+        }
+      }
+    } catch (Exception e) {
+      return StringUtils.EMPTY;
```

When we finally fail here, we hide the error. Wouldn't it be better to at least log the exception at this time? If there's another reason to not log it here, please leave a comment like there is on the other catch above.